### PR TITLE
Bug fix/collect aggregate undefined

### PR DESCRIPTION
**Pull request by hkernbach**
http://jenkins01:8080/view/PR/job/arangodb-matrix-pr/4231/

### CODE REVIEW

File: tests/js/server/aql/aql-optimizer-collect-count.js [hunk 1]
```diff
@@ -379,6 +379,21 @@ function optimizerCountTestSuite () {
         var plan = AQL_EXPLAIN(query).plan;
         assertNotEqual(-1, plan.rules.indexOf("collect-in-cluster"));
       }
+    },
+
+    testCollectAggregateUndefined: function () {
+      var randomDocumentID = db["UnitTestsCollection"].any()._id;
+      print(randomDocumentID);
```
Comment (mchacki) [inline tests/js/server/aql/aql-optimizer-collect-count.js]: please remove debug output

File: tests/js/server/aql/aql-optimizer-collect-count.js [hunk 2]
```diff
@@ -379,6 +379,21 @@ function optimizerCountTestSuite () {
...
+      var query = 'LET start = DOCUMENT("' + randomDocumentID + '")._key for i in [] collect aggregate count = count(i) return {count, start}';
+      var bindParams = {};
+      var options = {optimizer: {rules: ['-remove-unnecessary-calculations','-remove-unnecessary-calculations-2']}};
+
+      var results = db._query(query, bindParams, options).toArray();
+      // expectation is that we exactly get one result
+      // count will be 0, start will be undefined
+      assertEqual(1, results.length);
```
Comment (mchacki) [inline tests/js/server/aql/aql-optimizer-collect-count.js]: if the randomDocumentID is relevant you can add it here as a third parameter
assertEqual(1, results.length, query) which will print the failing query.

**Comment by hkernbach** [comment]
http://jenkins01:8080/view/PR/job/arangodb-matrix-pr/4284/

**Comment by hkernbach** [comment]
http://jenkins01:8080/view/PR/job/arangodb-matrix-pr/4296/

**Comment by hkernbach** [comment]
https://jenkins.arangodb.biz/job/arangodb-matrix-pr-linux/5827/

**Pull request merged**